src/fights.ts

```diff
@@ -1400,39 +1402,89 @@ const freeFightSources = [
+
+      // Consider forcing noncombats below:
+      if (globalOptions.prefs.yachtzeechain) return false; // NCs are better when yachtzeeing, probably
+      // TODO: With the KoL update, is there a function for checking if an NC is already forced?
+      if (have($item`Clara's bell`) && !globalOptions.clarasBellClaimed) {
+        return true;
+      }
+
+      // TODO: Calculate this against using the +5 fam weight buff
+      if (have($item`Cincho de Mayo`) && useableCinch() >= 60) {
+        return true;
+      }
+      return false; // It costs turns to do anything else here
     },
     () => {
-      if (have($item`Rufus's shadow lodestone`)) setChoice(1500, 2);
+      if (have($item`Rufus's shadow lodestone`)) {
+        setChoice(1500, 2); // Turn in lodestone if you have it
+        adv1(bestShadowRift(), -1, "");
+      }
       if (!get("_shadowAffinityToday") && !ClosedCircuitPayphone.rufusTarget()) {
         ClosedCircuitPayphone.chooseQuest(() => 2); // Choose an artifact (not supporting boss for now)
       }
-      adv1(bestShadowRift(), -1, "");
 
-      if (get("encountersUntilSRChoice", 0) === 0) {
-        if (ClosedCircuitPayphone.have() && !ClosedCircuitPayphone.rufusTarget()) {
-          ClosedCircuitPayphone.chooseQuest(() => 2);
+      // Potentially force an NC
+      let ncForced = get("encountersUntilSRChoice", 1) === 0;
+      if (!ncForced && !globalOptions.prefs.yachtzeechain && get("rufusQuestType") !== "items") {
+        if (have($item`Clara's bell`) && !globalOptions.clarasBellClaimed) {
+          globalOptions.clarasBellClaimed = true;
+          use($item`Clara's bell`);
+          ncForced = true;
+          // Jurassic parka
+        } else if (
+          have($item`Jurassic Parka`) &&
+          get("_spikolodonSpikeUses") < 5 &&
+          have($effect`Shadow Affinity`)
+        ) {
+          freeFightOutfit(new Requirement([], { forceEquip: $items`Jurassic Parka` }));
+          cliExecute("parka spikolodon");
+          const macro = Macro.skill($skill`Launch spikolodon spikes`).basicCombat();
+          // Adventure until we use spikes
+          const startingSpikes = get("_spikolodonSpikeUses");
+          do {
+            garboAdventureAuto(bestShadowRift(), macro);
+          } while (get("_spikolodonSpikeUses") === startingSpikes);
+          ncForced = true;
+        } else if (have($item`Cincho de Mayo`) && useableCinch() >= 60) {
+          while (get("_cinchUsed", 0) > 40) {
+            if (!freeRest()) throw new Error("We are out of free rests!");
+          }
+          useSkill($skill`Cincho: Fiesta Exit`);
+          ncForced = true;
         }
-        adv1(bestShadowRift(), -1, ""); // grab the NC
       }
 
+      // If you forced an NC, make sure you have a quest to do
+      if (ncForced && ClosedCircuitPayphone.have() && !ClosedCircuitPayphone.rufusTarget()) {
+        ClosedCircuitPayphone.chooseQuest(() => 2);
+      }
+
+      adv1(bestShadowRift(), -1, "");
```

Comment:
I'd like to rethink the structure of this.

It was previously much clearer which `adv1`s were for fighting the free rift mobs, and which `adv1`s were for grabbing the NCs. Both are now mixed into this particular `adv1` here, which feels weird. 

Especially because the rough structure I have in my mind is that we should be passing the Spikes macro here (when intended) - i.e. when we `adv1` to fight a free rift mob, we decide whether we should spike in this particular `adv1` (and then go on to handle the NC if we did decide to spike, which isn't too different from how we were previously handling the scheduled NC if it was coming up next).

Maybe Phred would have other ideas to chip in.

Reply:
I think this will probably need to be sliced out into several functions, possibly in its own file for simplicity. 

The exact way we proceed will vary a lot by whether or not the `Adventure Modifiers` thing gets added to api.php in the near future. If we're expecting it any day now, I'd just wait until we can definitively track whether an NC is up.

Reply:
Let's do it like this.
```
() => {
      if (have($item`Rufus's shadow lodestone`)) setChoice(1500, 2);
      if (!get("_shadowAffinityToday") && !ClosedCircuitPayphone.rufusTarget()) {
        ClosedCircuitPayphone.chooseQuest(() => 2); // Choose an artifact (not supporting boss for now)
      }
      runShadowRiftTurn(); // was previously adv1(bestShadowRift(), -1, ""); 
      if (get("encountersUntilSRChoice", 0) === 0) {
        if (ClosedCircuitPayphone.have() && !ClosedCircuitPayphone.rufusTarget()) {
          ClosedCircuitPayphone.chooseQuest(() => 2);
        }
        adv1(bestShadowRift(), -1, ""); // grab the NC
      }

      if (questStep("questRufus") === 1) {
        withChoice(1498, 1, () => use($item`closed-circuit pay phone`));
      }

      if (!have($effect`Shadow Affinity`) && get("encountersUntilSRChoice") !== 0) {
        setLocation($location`Friar Ceremony Location`); // Reset location to not affect mafia's item drop calculations
      }
    },
```

We then have the function
```
function runShadowRiftTurn(): void { // we can probably have a better name
  if (get("encountersUntilSRChoice") === 0) return;
  if (globalOptions.prefs.yachtzeechain || get("rufusQuestType") === "items") {
    adv1(bestShadowRift(), -1, ""); // We shouldn't be using NC forcers
    return;
  }

  if (have($item`Clara's bell`) && !globalOptions.clarasBellClaimed) {
    globalOptions.clarasBellClaimed = true;
    use($item`Clara's bell`);
  } else if (have($item`Cincho de Mayo`) && useableCinch() >= 60) {
      equip($slot`acc3`, $item`Cincho de Mayo`);
      while (get("_cinchUsed", 0) > 40) {
        if (!freeRest()) throw new Error("We are out of free rests!");
      }
      useSkill($skill`Cincho: Fiesta Exit`);
      // we should probably re-equip whatever was previously on $slot`acc3` here
  } else if (
      have($item`Jurassic Parka`) &&
      get("_spikolodonSpikeUses") < 5 &&
      have($effect`Shadow Affinity`) &&
      get("encountersUntilSRChoice") >= 2
    ) {
      freeFightOutfit(new Requirement([], { forceEquip: $items`Jurassic Parka` }));
      cliExecute("parka spikolodon");
      const macro = Macro.skill($skill`Launch spikolodon spikes`).basicCombat();
      garboAdventureAuto(bestShadowRift(), macro);
  } else {
    adv1(bestShadowRift(), -1, ""); // We wanted to use NC forcers, but none are suitable now
  }
}
```